### PR TITLE
OpenMote: fixed antenna driver

### PR DESCRIPTION
**Pull request by 9morano**
The drivers have been generalized so that they can be used on both OpenMote-CC2538 and OpenMote-B boards.
By default, OpenMote-CC2538 will use external antenna connector, and OpenMote-B will use the CC2538 radio.
This PR also fixes the incorrect use of the `GPIO_WRITE_PIN` macro.

### Detailed

Both OpenMote-CC2538 and OpenMote-B boards contain an antenna switch.
- For OpenMote-CC2538, the switch is used to connect the output of the radio to either the internal (chip antenna) or external antenna (SMA connector).
- For OpenMote-B, the switch selects which output of the radio goes to the SMA connector - either the Atmel's AT86RF215 radio or the TI CC2538 radio.

### CODE REVIEW

**Comment by kkrentz** [comment]
Thank you. This works on OpenMote-B.

File: arch/platform/openmote/dev/antenna.c [hunk 1]
```diff
@@ -52,25 +58,27 @@ void
...
-  GPIO_WRITE_PIN(ANTENNA_BSP_RADIO_BASE, ANTENNA_BSP_RADIO_EXT, 1);
 }
 /*---------------------------------------------------------------------------*/
 void
-antenna_internal(void)
+antenna_select(uint8_t sel)
 {
-  GPIO_WRITE_PIN(ANTENNA_BSP_RADIO_BASE, ANTENNA_BSP_RADIO_EXT, 0);
-  GPIO_WRITE_PIN(ANTENNA_BSP_RADIO_BASE, ANTENNA_BSP_RADIO_INT, 1);
+  if(sel == 0){
```
Comment (pjonsson) [inline arch/platform/openmote/dev/antenna.c:70]: Code formatting: `if(sel == 0) {`

File: arch/platform/openmote/dev/antenna.c [hunk 2]
```diff
@@ -52,25 +58,27 @@ void
...
+    /* On OpenMote-CC2538, select external antenna
+     * On OpenMote-B     , select AT86RF215 */
+    GPIO_CLR_PIN(ANTENNA_BSP_RADIO_BASE, ANTENNA_BSP_RADIO_INT);
+    GPIO_SET_PIN(ANTENNA_BSP_RADIO_BASE, ANTENNA_BSP_RADIO_EXT);
+  }
```
Comment (pjonsson) [inline arch/platform/openmote/dev/antenna.c:75]: Code formatting: `} else {`

If you have `uncrustify` installed and run `tools/code-style/uncrustify-check-style.sh file.c`, you will get a diff for lines that do not follow the code style. Don't worry about lines you haven't changed.

File: arch/platform/openmote/dev/antenna.h [hunk 3]
```diff
@@ -32,36 +32,38 @@
...
+ *            a value of 0 will connect:
+ *             - OpenMote-CC2538: CC2538 radio to external antenna
+ *             - OpenMote-B     : SMA connector to AT86RF215 radio
+ *            any other value will connect:
+ *             - OpenMote-CC2538: CC2538 radio to internal antenna
+ *             - OpenMote-B     : SMA connector to CC2538 radio
  */
-void antenna_external(void);
-/*---------------------------------------------------------------------------*/
+void antenna_select(uint8_t sel);
```
Comment (pjonsson) [inline arch/platform/openmote/dev/antenna.h:66]: Is there a potential third value for `sel` in the future, or should the parameter be a `bool`?

**Comment by kkrentz** [comment]
What would go beyond just fixing is to introduce a self-explanatory and consistent naming. Currently, the antennas are referred to as internal/external, ANT1/ANT2, and true/false. To my mind, a better naming could be 2.4-GHz/sub-GHz. On the OpenMote-CC2538 platform, the sub-GHz antenna would be the one connected to the CC1200. On the OpenMote-B platform, the sub-GHz antenna would be the one connected to the AT86RF215. As the parameter to `antenna_select`, I would rather use an enum, even if there are only two possible options (2.4-GHz/sub-GHz). `sel` does not sound like a boolean.

**Comment by 9morano** [comment]
Are we talking about the same platform? If I am not mistaken, the [OpenMote-CC2538](https://www.cs.virginia.edu/~bjc8c/class/cs6501-f18/papers/vilajosana15openmote.pdf) does not include a sub-GHz interface (there is no CC1101 on board).

**Comment by kkrentz** [comment]
The CC1200 is part of OpenUSB 2.

**Comment by 9morano** [comment]
I didn't know that. Is there any RF switch on that **OpenUSB 2** platform?

Because I've looked through the _scarce_ documentation on OpenMote-CC2538 and I think that there is actually no RF switch on the board! I also went through the OpenMote-CC2538 platform ports on other operating systems (RIOT, OpenWSN) and nowhere is an antenna switch used or defined.

I only own an OpenMote-B, so if you could please help me with understanding, why did Contiki-NG had included those functions `antenna_select_external()` and `antenna_select_internal()` for OpenMote-CC2538?

**Comment by 9morano** [comment]
> On the OpenMote-B platform, the sub-GHz antenna would be the one connected to the AT86RF215. 

And to clarify, the sub-GHz part of AT86RF215 on OpenMote-B is always connected to the SMA connector. The RF switch on OpenMote-B is only used for 2.4 GHz part. 

OpenMote-B_RF_switch

**Comment by 9morano** [comment]
Great, thank you for taking the time to test it! :)
I will move the files to the openmote-b directory and remove unnecessary pin defines at openmote-cc2538.
In order not to pollute the commit history,  I will close this PR and create a new one

**Comment by pjonsson** [comment]
@9morano I haven't followed the discussion but you can edit the commit history freely in your local repository and then force-push to update PRs for Contiki-NG.

**Comment by kkrentz** [comment]
I successfully tested this version on both types of OpenMote platforms.

File: arch/platform/openmote/openmote-b/board.h [hunk 4]
```diff
@@ -189,9 +189,9 @@
  *
  * @{
  */
-#define ANTENNA_BSP_RADIO_BASE   GPIO_PORT_TO_BASE(GPIO_D_NUM)
-#define ANTENNA_BSP_RADIO_INT    GPIO_PIN_MASK(3)
-#define ANTENNA_BSP_RADIO_EXT    GPIO_PIN_MASK(4)
+#define ANTENNA_BSP_RADIO_BASE              GPIO_PORT_TO_BASE(GPIO_D_NUM)
+#define ANTENNA_BSP_RADIO_24GHZ_CC2538      GPIO_PIN_MASK(3)
+#define ANTENNA_BSP_RADIO_24GHZ_AT86RF215   GPIO_PIN_MASK(4)
```
[the remark this answers is not in the document]
Comment (9morano) [reply arch/platform/openmote/openmote-b/board.h:194]: Yes, the AT86RF215 chip has 2 radio cores: sub-GHz and 2.4 GHz.
By using this `_24GHZ_` part I tried to emphasize that only the 2.4 GHz part can be controlled with the RF switch ... the 868 MHz radio core is always connected to the 868 MHz SMA connector.

**Review by nvt — APPROVED** [review]

**Pull request merged**